### PR TITLE
fix gradle integration tests 

### PR DESCRIPTION
**Pull request by evasiba-ericsson**
I cannot do a squash due to the non linear history of my commits and merged commits from master.

### CODE REVIEW

**Comment by p-backman-ericsson** [comment]
I think it is easier if you always work on a different branches for each feature in your forked repo. Hopefully we can squash them when merging.

File: build.gradle [hunk 1]
```diff
@@ -146,8 +164,7 @@ dependencies {
     compile group: 'io.reactivex', name: 'rxjava', version: '1.1.5'*/
 
     // In order to get embedded app container to work
-    integrationTestRuntime("org.springframework.boot:spring-boot-starter-tomcat")
-    integrationTestCompile group: 'com.jayway.restassured', name: 'rest-assured', version: '2.9.0'
+//    integrationTestRuntime("org.springframework.boot:spring-boot-starter-tomcat")
```
Comment (p-backman-ericsson) [inline build.gradle]: Remove commented away line

File: src/integration-test/java/com/ericsson/eiffel/remrem/generate/EiffelSemanticsController.java [hunk 2]
```diff
@@ -58,7 +56,7 @@ public void  setUp() throws FileNotFoundException {
             then().
             statusCode(HttpStatus.SC_OK)
             .body("meta.type", Matchers.is("eiffelartifactpublished"))
-            .body("meta.version", Matchers.is("0.1.5"));
+            .body("meta.version", Matchers.is("0.1.7"));
```
Comment (p-backman-ericsson) [inline src/integration-test/java/com/ericsson/eiffel/remrem/generate/EiffelSemanticsController.java]: Would it be possible to use the information specified in build.gradle?

File: src/main/java/com/ericsson/eiffel/remrem/generate/cli/CLI.java [hunk 3]
```diff
@@ -41,11 +42,17 @@
     @Autowired
     private MsgService[] msgServices;
 
-    @Override
+    public CLI(MsgService[] msgServices) {
+		super();
```
Comment (p-backman-ericsson) [inline src/main/java/com/ericsson/eiffel/remrem/generate/cli/CLI.java]: Our standard is to use 4 spaces to indent. Not a tab.

File: src/main/java/com/ericsson/eiffel/remrem/generate/cli/CLIExitCodes.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+	public static int CLI_EXCEPTION=1;
+	public static int CLI_MISSING_OPTION_EXCEPTION=2;
+	public static int CLI_READ_FILE_FAILED=3;
+	public static int READ_JSON_FROM_CONSOLE_FAILED=4;
+	public static int HANDLE_JSON_STRING_FAILED=5;
+	public static int MESSAGE_PROTOCOL_NOT_FOUND=6;
+	
+	public static int getExceptionCode(Exception e) {
+		if (e instanceof MissingOptionException)
+        	return CLI_MISSING_OPTION_EXCEPTION;
```
Comment (p-backman-ericsson) [inline src/main/java/com/ericsson/eiffel/remrem/generate/cli/CLIExitCodes.java]: Without indentation this code is hard to read. Please add correct indentation and curly brackets {} to improve readability.

File: src/test/java/com/ericsson/eiffel/remrem/generate/cli/CliOptionsUnitTests.java [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+		int code = CLIExitCodes.CLI_MISSING_OPTION_EXCEPTION;
+		assertTrue(CLIOptions.getErrorCodes().contains(code));
+	}
+	
+	@Test
+	public void testParseAllCLIOptions() throws Exception {
+		String[] args = {"-t", "mt", "-f", "filename"};	    
+
+		CLIOptions.parse(args);
+		int i = 0;
```
Comment (p-backman-ericsson) [inline src/test/java/com/ericsson/eiffel/remrem/generate/cli/CliOptionsUnitTests.java]: This doesn't assert anything

File: src/integration-test/java/com/ericsson/eiffel/remrem/generate/EiffelSemanticsController.java (hunk 2 above)
Comment (evasiba-ericsson) [inline src/integration-test/java/com/ericsson/eiffel/remrem/generate/EiffelSemanticsController.java]: no. but we can read it from manifest file of semantics. I will update it tomorrow

**Comment by evasiba-ericsson** [comment]
tried to squash the last two commits but failed again. We got 3 more commits instead of only one commit

**Pull request merged**